### PR TITLE
Eager share only core packages and their dependencies

### LINKED ISSUE #9329: Shared module is not available for eager consumption
Testing with rc8 and a mix of federated and non-federated extensions.

The federated extension shares its copy of the shared package metadata with:

```json
  "sharedPackages": {
      "mypackage": {
        "requiredVersion": "^0.6.0",
        "bundled": true,
        "singleton": true
      }
    }
```

The non-federated extension defines the shared package metadata with:

```json
"sharedPackages": {
      "mypackage": {
        "requiredVersion": "^0.6.0",
        "bundled": false,
        "singleton": true
      }
    }
```

Would yield the following in the browser console:

```
Error: Shared module is not available for eager consumption: webpack/sharing/consume/default/mypackage
```

(see https://github.com/jtpio/jupyterlab-topbar/pull/53 for more details)

It looks like it may be related to the eager mode added in https://github.com/jupyterlab/jupyterlab/pull/9312

### PR DESCRIPTION
**Pull request by jasongrout**
## References

Fixes #9329 

## Code changes

Dials back the eager sharing done in #9312 to just core packages

### CODE REVIEW

**Review by blink1073 — APPROVED** [review]
Thanks!

**Comment by jtpio** [comment]
Just tested the patch locally, checking the generated `build_log.json` which contains the `eager` key only for core packages. For example:

```json
"shared": {
	"@jupyterlab/application": {
	  "requiredVersion": "~3.0.0-rc.8",
	  "eager": true,
	  "singleton": true
	},
	"@jupyterlab/application-extension": {
	  "requiredVersion": "~3.0.0-rc.8",
	  "eager": true
        },
```

Looks good!

**Comment by jasongrout** [comment]
Agh, linting is the bane of my PRs! I'm going to have to do something about the setup of my editor (I thought we had a linting check done on every commit!)

**Comment by jtpio** [comment]
> I thought we had a linting check done on every commit!

It might be related to this: https://github.com/jupyterlab/jupyterlab/issues/9106

**Comment by jasongrout** [comment]
> It might be related to this: #9106

Ah, that would explain it!

**Comment by jasongrout** [comment]
I've uninstalled my system yarn to keep things more isolated, and I commonly use nodejs installed in a conda environment.

File: dev_mode/webpack.config.js [hunk 1]
```diff
@@ -160,49 +160,68 @@ function ignored(path) {
...
+    if (eagerPackages.has(pkg)) {
+      shared[pkg].eager = true;
+    }
   }
 }
 
 // Add dependencies and sharedPackage config from extension packages if they
 // are not already in the shared config. This means that if there is a
 // conflict, the resolutions package version is the one that is shared.
 extraShared = [];
```
Comment (jtpio) [inline dev_mode/webpack.config.js:193]: Speaking of linting, looks like this var is missing a const:

```suggestion
const extraShared = [];
```

Comment (jasongrout) [reply dev_mode/webpack.config.js]: Weird that lint didn't pick that up.

Comment (jasongrout) [reply dev_mode/webpack.config.js]: (or maybe it did as a warning, one of the hundreds of warnings that would be good to sprint on)

**Comment by jasongrout** [comment]
Cutting rc9 with this now...

**Comment by jtpio** [comment]
Thanks!

**Pull request merged**